### ISSUE TITLE
Can not use custom type as a query parameter

### ISSUE THREAD

**Issue by defusioner** [Akryum/vue-apollo#361]
Hello there!

Can't get around with the error, can't event find an adequate answer on Google:
`Error in created hook: "Error: Schema type definitions not allowed in queries. Found: "InputObjectTypeDefinition""`

This happens when I'm trying to run a query with a custom `input` parameter: 

queries.graphql
```
#import "./fragments.graphql"

query labels($opts: SearchCriteriaInput) {
	labels(opts: $opts) {
		...LabelsFragment
	}
}
```

fragments.graphql
```
#import "./types.graphql"
fragment LabelsFragment on Label {
	name
}
```

types.graphql
```
input SearchCriteriaInput {
  filterField: String
}

type Label {
  name: String
}
```

I have a `graphql-yoga` back-end, this request `works well in graphql-playground`, but my front-end component can't perform it at all. 

Something that I miss? 

Thanks

**Comment by Akryum** [comment]
You can't define types in the queries, as the error message says. They belong to the type schema.

**Comment by defusioner** [comment]
@Akryum Any reason for this? I'm looking at the reference of GraphQL, it says that I do can use the custom type: https://graphql.org/learn/queries/#directives

**Comment by Akryum** [comment]
It doesn't change the fact that types must be defined in the server schema.

**Comment by defusioner** [comment]
@Akryum OK, so the workaround is to define a real input on the back-end, and a JSON tupple on the client. Why can't queries reuse custom inputs as mutations do? This looks logical for me.

**Comment by defusioner** [comment]
We are building an API that is some sort of Data-Driven model, so we define one schema and generate the API for both back-end and front-end. It sound strange that front-end can't see the input given :)

**Comment by Akryum** [comment]
The client can use all the types and inputs defined on the server schema, I'm not sure I understand why you absolutely want to define them on client-side.

**Comment by defusioner** [comment]
@Akryum Thanks to `GraphQL` slack chat I realized my problem.

So yes, we are making an API that generates `types`, `queries`, `mutation` for the back, and `fragments`, `queries`, `mutations` for the front. That simplifies our microservices data schema (and lets us reuse some data schemas inside another one bigger). 

As we use Vue-CLI, and, obviously, webpack with graphql-loader, we can import these types via `#import` tag (I don't know where I found it, but it worked). 

So, when I was trying to run this `front-end query`, I was importing my fragments, when they're importing the `types`, generated for the back-end. -> The source of the problem.

I was thinking that Apollo should know the type that is extended / used by the fragment, but we can literally plug to this "hard-coded string", and define our fragment structure -> this lets not to import types.

Thanks! Declarative programming is the best.
